core/src/main/groovy/com/novoda/gradle/release/GradlePluginPropertyFinder.groovy

```diff
@@ -0,0 +1,37 @@
+package com.novoda.gradle.release
+
+import org.gradle.api.Project
+import org.gradle.api.file.FileCollection
+import org.gradle.api.file.FileTree
+
+class GradlePluginPropertyFinder {
+    private static final String FILE_EXTENSION_PROPERTIES = ".properties"
+    private final Project project
+
+    GradlePluginPropertyFinder(Project project) {
+        this.project = project
+    }
+
+    public String findBestGradlePluginId() {
+        FileTree pluginFiles = project.fileTree(dir:'src/main/resources/META-INF/gradle-plugins')
+        if (!pluginFiles.isEmpty()) {
```

Comment:
same here: Early return increases readability and clears the requirements for executing the rest of the method body.